File: Tameenk/src/app/core/models/product-benefit.model.ts
import { IBenefit } from "./benefit.model";

export class IProductBenefit {
  id: number;
  productId: string;
  benefitId: number;
  isSelected: boolean;
  benefitPrice: number;
  benefitExternalId: string;
  isReadOnly: boolean;
  benefit: IBenefit;
  isDisabled: boolean;
  constructor() {}
}

File: Services/Quotation/Tameenk.Services.QuotationApi/Models/ProductBenefitModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Tameenk.Services.QuotationApi.Models
{
    [JsonObject("productBenefit")]
    public class ProductBenefitModel
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("productId")]
        public Guid? ProductId { get; set; }

        [JsonProperty("benefitId")]
        public short? BenefitId { get; set; }

        [JsonProperty("isSelected")]
        public bool? IsSelected { get; set; }

        [JsonProperty("benefitPrice")]
        public decimal? BenefitPrice { get; set; }

        [JsonProperty("benefitExternalId")]
        public string BenefitExternalId { get; set; }

        [JsonProperty("isReadOnly")]
        public bool IsReadOnly { get; set; }

        [JsonProperty("benefit")]
        public BenefitModel Benefit { get; set; }

        
    }
}